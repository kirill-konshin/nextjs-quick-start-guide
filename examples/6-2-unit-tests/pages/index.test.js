import {shallow} from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';
import Index from './index.js'

describe('Enzyme', () => {

    it('Renders "Hello octocat!" for given props', () => {
        const app = shallow(<Index userinfo={{login: 'octocat'}}/>);
        expect(app.find('div').text()).toEqual('Hello, octocat!');
    });

});

describe('Snapshot Testing', () => {

    it('Renders "Hello octocat!" for given props', () => {
        const component = renderer.create(<Index userinfo={{login: 'octocat'}}/>);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Renders "Hello octocat!" for emulated NextJS lifecycle', async () => {
        const userinfo = Index.getInitialProps({});
        const component = renderer.create(<Index userinfo={userinfo}/>);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

});