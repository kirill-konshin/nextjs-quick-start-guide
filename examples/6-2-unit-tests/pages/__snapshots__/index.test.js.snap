// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Testing Renders "Hello octocat!" for emulated NextJS lifecycle 1`] = `
<div>
  Hello, 
  !
</div>
`;

exports[`Snapshot Testing Renders "Hello octocat!" for given props 1`] = `
<div>
  Hello, 
  octocat
  !
</div>
`;
